src/naive/date.rs

```diff
@@ -596,6 +596,59 @@ impl NaiveDate {
         parsed.to_naive_date()
     }
 
+    /// An addition of months to `NaiveDate` clamped to valid days in resulting month.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use chrono::{Duration, NaiveDate};
+    ///
+    /// let from_ymd = NaiveDate::from_ymd;
+    ///
+    /// assert_eq!(from_ymd(2014, 1, 1).add_months(1),                  from_ymd(2014, 2, 1));
+    /// assert_eq!(from_ymd(2014, 1, 31).add_months(1),                 from_ymd(2014, 2, 28));
+    /// assert_eq!(from_ymd(2020, 1, 31).add_months(1),                 from_ymd(2020, 2, 29));
+    /// assert_eq!(from_ymd(2014, 1, 1).add_months(-11),                from_ymd(2013, 2, 1));
+    /// assert_eq!(from_ymd(2014, 1, 1).add_months(-12),                from_ymd(2013, 1, 1));
+    /// assert_eq!(from_ymd(2014, 1, 1).add_months(-13),                from_ymd(2012, 12, 1));
+    /// ```
+    pub fn add_months(&self, months: i32) -> NaiveDate {
+        let target = self.add_month_get_first_day(months);
+        let target_plus = target.add_month_get_first_day(1);
+        let last_day = target_plus.sub(Duration::days(1));
+        let day = core::cmp::min(self.day(), last_day.day());
+        NaiveDate::from_ymd(target.year(), target.month(), day)
+    }
+
+    /// Private function to calculate necessary primitives for `add_months()`
+    ///
+    /// # Arguments
+    ///
+    /// * `delta` - Number of months (+/-) to add
+    ///
+    /// # Returns
+    ///
+    /// A new NaiveDate on the first day of the resulting year & month
+    ///
+    fn add_month_get_first_day(&self, delta: i32) -> NaiveDate {
```

Comment:
if possible can you add a few test cases for this function

Reply:
I couldn't put them in the docs since it's private, so I added them to the test module.